### PR TITLE
remove install_ext

### LINKED ISSUE #8634: remove %install_ext
`%install_ext` is now deprecated.  For 5.0 we can completely remove all reference of it. 

(marking as sprint friendly)

### PR DESCRIPTION
**Pull request by Achifaifa**
Deleted the function, checked for function references through code and tests (nothing found) and added information of the changes to the /docs/source/whatsnew/pr/ folder.

Not sure if the format used in the change docs is the correct one, the examples I checked were empty files, please check that before merging.

closes #8634

### CODE REVIEW

**Comment by Achifaifa** [comment]
OK, I've checked the tests in the IPython folder and didn't find any calls to install_ext in them, so I'm not sure where that call travis is making is or how to remove it. Working on it.

**Comment by Achifaifa** [comment]
Right! I missed that one. There is a extension test function that depended on install_ext. I've commented it out instead of deleting it just in case someone needs to reuse that code, since it didn't seem totally expendable. 

And sorry about the mess with the rst files, took me a while to figure those out.

**Comment by JanSchulz** [comment]
What is the replacement story? Build a package for each 10-liner? Would it be possible to add a ipython-contrib/ipython-extensions repo (with some guidelines what to add), which is released every so often?

**Comment by Carreau** [comment]
> What is the replacement story? Build a package for each 10-liner?

Yes, it's not that hard :

https://github.com/Carreau/pip_magic/blob/master/pip_magic/__init__.py

Package with [flit](https://pypi.python.org/pypi/flit) that make publishing packages really easy.

> Would it be possible to add a ipython-contrib/ipython-extensions

ipython-contrib is not an official IPython repo, but I thing the people there wouldn't say no. 

It would be released as often as the maintainers there would release, we could even set up travis to release really often.

**Comment by Carreau** [comment]
@Achifaifa we are now on the 5.0 branch, do you want to rebase, so that we can merge. You also directly delete the test instead of just commenting.

**Comment by Achifaifa** [comment]
Oops, missed the notification and didn't read this for some time.

File: IPython/core/tests/test_magic.py [hunk 1]
```diff
@@ -604,27 +604,6 @@ def test_prun_quotes():
...
-    sys.path.insert(0, daft_path)
-    try:
-        _ip.user_ns.pop('arq', None)
-        invalidate_caches()   # Clear import caches
-        _ip.magic("load_ext daft_extension")
-        nt.assert_equal(_ip.user_ns['arq'], 185)
-        _ip.magic("unload_ext daft_extension")
-        assert 'arq' not in _ip.user_ns
-    finally:
-        sys.path.remove(daft_path)
```
Comment (takluyver) [inline IPython/core/tests/test_magic.py]: Why delete this test? It doesn't seem to use `install_ext` at all.

**Comment by Achifaifa** [comment]
I removed the function I commented in a previous commit but apparently someone edited it somewhere along the way to not use install_ext, sorry about that.

https://github.com/Achifaifa/ipython/commit/fda188e316a8659b8f4e24aec05649d218997e99

**Comment by takluyver** [comment]
Aha, gotcha. The code as it stands reverts the test to the version that did use `%install_ext` - I'm going to rebase this, clean it up and merge manually.

**Comment by Achifaifa** [comment]
Oh great, thank you.

**Comment by Carreau** [comment]
> Oops, missed the notification and didn't read this for some time.

No problem, we have plenty of time.

> apparently someone edited it somewhere along the way to not use install_ext, sorry about that.

That SO was likely me to avoid deprecation warnings of using install ext :-( 


@takluyver said :
>  I'm going to rebase this, clean it up and merge manually.

FYI that this is because we attempt to avoid merge from the master branch usually,
but rebasing can be a pain to do. Thanks for doing it !

**Comment by takluyver** [comment]
Merged as d953069d49673af0fb5a69b3bdbe3a761181ac7f . Thanks @Achifaifa!

**Pull request closed without merging**